### PR TITLE
feat(*): Cleaner task running

### PR DESCRIPTION
**Pull request by thomastaylor312**
Puts together something for actually doing work and managing several loops. This makes it easier to encapsulate the actual work we have to do down the line

### CODE REVIEW

**Comment by thomastaylor312** [comment]
Until #60 merges, just filter this PR to the single commit to see what I did

**Comment by thomastaylor312** [comment]
Ignore build failures since I accidentally started with this against main

File: src/consumers/manager.rs [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    ///
+    /// The given work function should attempt to handle the event
+    pub async fn add_for_lattice<W>(&self, topic: &str, worker: W) -> Result<(), async_nats::Error>
+    where
+        W: Worker + Send + Sync + 'static,
+        C: Stream<Item = Result<ScopedMessage<W::Message>, async_nats::Error>>
+            + CreateConsumer<Output = C>
+            + Send
+            + Unpin
+            + 'static,
```
Comment (brooksmtownsend) [inline src/consumers/manager.rs:107]: tenor-108401073

**Review by brooksmtownsend — APPROVED** [review]
So this PR introduces the idea of "work" which I understand to be processing an event or command. As these are picked up on the NATS topic, we use a semaphore and create a work handler to represent the work needed to handle that event/command. With a slow stream of these this won't really change much, but with a large stream of events or commands the ConsumerManager will essentially deal with dealing out permits to let them execute.

Is this a decent summary? If this is way off the mark, then my approval may be moot

**Comment by thomastaylor312** [comment]
Yep, you have it exactly right

**Pull request merged**